app/ui/src/app/integration/edit-page/current-flow.service.ts

```diff
@@ -349,6 +349,15 @@ export class CurrentFlowService {
     return (position + 1) >= this.steps.length;
   }
 
+  isActionShapeless(descriptor: ActionDescriptor) {
+    if (!descriptor) {
+      return false;
+    }
+    const inputDataShape = descriptor.inputDataShape;
+    const outputDataShape = descriptor.outputDataShape;
```

Comment:
Just a suggestion for the future. You can streamline these kinda assignations by harnessing the power of ES6 destructuring:

```javascript
const { inputDataShape, outputDataShape } = descriptor;
```